boards/samr21-xpro/Makefile.include

```diff
@@ -2,7 +2,29 @@
 export CPU = samd21
 export CPU_MODEL = samr21g18a
 
-# set edbg device type
-EDBG_DEVICE_TYPE = atmel_cm0p
+## setup the boards dependencies
+include $(RIOTBOARD)/$(BOARD)/Makefile.dep
 
-include $(RIOTMAKE)/boards/sam0.inc.mk
+export PROGRAMMER ?= edbg
+
+ifeq ($(PROGRAMMER),edbg)
+  # set edbg device type
+  EDBG_DEVICE_TYPE = atmel_cm0p
+
+  include $(RIOTMAKE)/boards/sam0.inc.mk
+else ifeq ($(PROGRAMMER),jlink)
+  export DEBUGGER = $(RIOTBOARD)/$(BOARD)/dist/debug.sh
+  export DEBUGSERVER = JLinkGDBServer -device atsamr21e18a -if swd
+  export RESET = $(RIOTBOARD)/$(BOARD)/dist/reset.sh
+
+  export FLASHER = $(RIOTBOARD)/$(BOARD)/dist/flash.sh
+  export FFLAGS  = $(BINDIR) $(HEXFILE)
+
+  export OFLAGS    = -O binary --gap-fill 0xff
+  export HEXFILE = $(ELFFILE:.elf=.bin)
+  export DEBUGGER_FLAGS = $(BINDIR) $(ELFFILE)
+  export RESET_FLAGS = $(BINDIR)
+
+  export TERMPROG = JLinkRTTClient
```

Comment:
Using rtt_stdio is separate from flashing with JTAG, you may want to flash with JTAG but still use the serial for stdio

Reply:
Maybe put a guard here and only specify this if rtt_stdio is in USE_MODULEs